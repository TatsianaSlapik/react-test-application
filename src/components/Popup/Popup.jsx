import React from "react";

import "./Popup.scss";

export default function Popup({ onHidePopup }) {
  return (
    <div className="popup">
      <div className="popup_inner">
        <h2 className="popup_text">Product added to cart</h2>
        <button className="popup_btn" onClick={onHidePopup}>
          ОК
        </button>
      </div>
    </div>
  );
}
