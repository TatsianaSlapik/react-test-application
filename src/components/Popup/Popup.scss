.popup {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: rgba(0, 0, 0, 0.75);
}

.popup_inner {
  display: flex;
  align-items: center;
  flex-direction: column;
  position: absolute;
  border-radius: 8px;
  left: 25%;
  right: 25%;
  top: 25%;
  bottom: 25%;
  margin: auto;
  background: white;
  justify-content: center;
}

.popup_text {
  text-align: center;
  margin-bottom: 50px;
}

.popup_btn {
  border: 4px solid #ff1100;
  border-radius: 8px;
  padding: 5px 10px;
  color: #ffffff;
  font-size: 20px;
  background-color: #ff1100;
  width: 150px;
  margin: 0 auto;
  &:hover {
    cursor: pointer;
  }
}
