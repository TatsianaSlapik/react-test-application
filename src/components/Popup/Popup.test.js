import React from "react";
import renderer from "react-test-renderer";
import "babel-polyfill";
import Popup from "./Popup.jsx";
import { Provider } from "react-redux";

import configureStore from "redux-mock-store";
const mockStore = configureStore([]);
describe("Popup", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      showAddToCartPopup: true,
    });
    component = renderer.create(
      <Provider store={store}>
        <Popup />
      </Provider>
    );
  });

  it("Popup when showPopup is true should be rendered", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
