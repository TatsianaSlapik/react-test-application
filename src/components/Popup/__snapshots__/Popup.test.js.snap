// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Popup when showPopup is true should be rendered 1`] = `
<div
  className="popup"
>
  <div
    className="popup_inner"
  >
    <h2
      className="popup_text"
    />
    <button
      className="popup_btn"
      onClick={[Function]}
    >
      ОК
    </button>
  </div>
</div>
`;
